toolkit/FileManager.cpp

```diff
@@ -0,0 +1,62 @@
+#include "FileManager.hpp"
+
+#include <thread>
+#include <string.h>
+#include "base/Logger.hpp"
+
+/// \namespace darwin
+namespace darwin {
+    /// \namespace toolkit
+    namespace toolkit{
+
+        FileManager::FileManager(std::string& file, bool app)
+                : app{app}, file{file} {}
+
+        bool FileManager::Open() {
+            if(file_stream.is_open()){
+                return true;
+            }
+
+            file_stream.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+            try {
+                if(app){
+                    file_stream.open(file, std::fstream::app);
+                }
+                else{
+                    file_stream.open(file);
+                }
+            }catch (std::fstream::failure& e) {
+                std::cerr << "Error when opening file..." << e.what();
+                return false;
+            }
+            return file_stream.is_open();
+        }
+
+        bool FileManager::Write(std::string s){
```

Comment:
```suggestion
        bool FileManager::Write(std::string& s){
```
We can use a reference here